sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -163,27 +165,76 @@ AccessToken TokenCredentialImpl::ParseToken(
 
     AccessToken accessToken;
     accessToken.Token = parsedJson[accessTokenPropertyName].get<std::string>();
+    accessToken.ExpiresOn = std::chrono::system_clock::now();
 
     if (parsedJson.contains(expiresInPropertyName))
     {
-      accessToken.ExpiresOn
-          = std::chrono::system_clock::now()
-          + std::chrono::seconds(
-                parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
+      try
+      {
+        // 'expires_in' as number (seconds until expiration)
+        accessToken.ExpiresOn += std::chrono::seconds(
+            parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
+
+        return accessToken;
+      }
+      catch (std::exception const&)
+      {
+        try
+        {
+          // 'expires_in' as numeric string (seconds until expiration)
```

Comment:
expires_on as numeric string?

Reply:
In other words, I think the comment here is incorrect.

Reply:
The comment is correct. We read `parsedJson[expiresInPropertyName]`.